IoTDeviceSimulator/azure-functions/d2c-simulator/index.js

```diff
@@ -17,28 +17,169 @@ module.exports = async function (context, req) {
   const iotHubName = ConnectionString.parse(process.env[req.body.connectionStringId], ["HostName"]).HostName;
   let deviceArray = req.body.selectedDevices.map((deviceTwin) => {
     return {
-
       deviceId: deviceTwin.deviceId,
       telemetrySendInterval: deviceTwin.telemetrySendInterval ?? 5000,
       unit: deviceTwin.unit,
-      mean: deviceTwin.mean,
-      amplitude: deviceTwin.amplitude,
       phenomenon: deviceTwin.phenomenon,
       valueIsBool: deviceTwin.valueIsBool,
       deviceSecurityType: "connectionString",
       connectionString: `HostName=${iotHubName};DeviceId=${deviceTwin.deviceId};SharedAccessKey=${deviceTwin.primaryKey}`,
       phase: Math.random() * Math.PI,
-      behaviour: deviceTwin.behaviour,
-      noise_magnitude: deviceTwin.noise_magnitude,
       noiseSd: deviceTwin.noiseSd,
-      sine_period: deviceTwin.sine_period,
       isRunning: deviceTwin.isRunning,
       min: deviceTwin.min,
       max: deviceTwin.max,
+      currDataArray: deviceTwin.currDataArray,
+      signalArray: deviceTwin.signalArray,
     }
   });
 
   const modelIdObject = { modelId: 'dtmi:com:example:Thermostat;1' };
+
+  class generator {
```

Comment:
Can we move all signal generator code into another file (lets say SignalGenerator.js)?